src/main/java/uk/gov/hmcts/reform/judicialbooking/domain/service/common/PrepareDataService.java

```diff
@@ -0,0 +1,88 @@
+package uk.gov.hmcts.reform.judicialbooking.domain.service.common;
+
+import org.apache.commons.lang.StringUtils;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.judicialbooking.controller.advice.exception.ResourceNotFoundException;
+import uk.gov.hmcts.reform.judicialbooking.data.BookingEntity;
+import uk.gov.hmcts.reform.judicialbooking.domain.model.Appointment;
+import uk.gov.hmcts.reform.judicialbooking.domain.model.Authorisation;
+import uk.gov.hmcts.reform.judicialbooking.domain.model.BookingResponse;
+import uk.gov.hmcts.reform.judicialbooking.domain.model.JudicialUserProfile;
+import uk.gov.hmcts.reform.judicialbooking.domain.model.OrmBooking;
+import uk.gov.hmcts.reform.judicialbooking.domain.model.OrmBookingAssignmentsRequest;
+import uk.gov.hmcts.reform.judicialbooking.domain.model.OrmBookingRequest;
+import uk.gov.hmcts.reform.judicialbooking.domain.model.enums.Status;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class PrepareDataService {
+
+    public BookingEntity prepareBooking(BookingEntity booking, List<JudicialUserProfile> judicialUserProfiles) {
+
+        List<Appointment> appointment = judicialUserProfiles.get(0).getAppointments().stream()
```

Comment:
What if the judicialUserProfiles list is empty? This can throw NPE.

Reply:
I think, there would be some validations as well on the retrieved judicialUserProfiles when we retrieve it from the actual JRD. There also we can validate the emptiness of list and throw some 422 error if no user profile found on the JRD we cannot proceed further.

Reply:
This method shouldn't deal with list of judicialUserProfile we can update the param to take single judicialUserProfiles object instead.